### PR TITLE
46 feat db connection

### LINKED ISSUE #46: 📝Create Supabase DB connection
# Supabase!

- [x] Create supabase account
- [x] Create basic user schema on supabase
- [x] Connect to supabase in the app
- [x] Build sample login screen (no styling)
- [x] Route to sample dashboard screen on success?

## 🔗 Resources

https://supabase.com/docs/guides/with-react

### PR DESCRIPTION
**Pull request by maxh1231**
- Supabase NPM package and DB -> Client connection
- SignupScreen
- LoginScreen
- Dashboard
- .env template file

Closes #46

### CODE REVIEW

File: src/screens/Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { supabase } from '../helpers/supabaseClient';
+import { useSession } from '../AppWrapper';
+
+const Dashboard = () => {
```
Comment (Thenlie) [inline src/screens/Dashboard.tsx:4]: Dashboard -> DashboardScreen

File: src/screens/Dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	const { session } = useSession();
+	console.log(session);
+
+	return (
+		<div aria-live="polite">
+			<form className="form-widget">
+				<div>Email: {session?.user.email}</div>
+			</form>
+			<button type="button" className="button block" onClick={() => supabase.auth.signOut()}>
```
Comment (Thenlie) [inline src/screens/Dashboard.tsx:13]: Where are these `button, block` classes defined?

File: src/AppWrapper.tsx [hunk 3]
```diff
@@ -1,19 +1,61 @@
-import { Outlet } from 'react-router-dom';
+import { Outlet, useOutletContext } from 'react-router-dom';
 import './App.css';
```
Comment (Thenlie) [inline src/AppWrapper.tsx:3]: nit: empty line, more below

File: src/screens/LoginScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+							type="email"
+							placeholder="Your email"
+							value={email}
+							onChange={(e) => setEmail(e.target.value)}
+						/>
+						<input
+							id=""
+							className="inputField"
+							type="password"
+							placeholder="Your pasword"
```
Comment (Thenlie) [inline src/screens/LoginScreen.tsx:39]: pasword -> password

File: src/screens/LoginScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react';
+import { supabase } from '../helpers/supabaseClient';
+
+export default function LoginScreen() {
+	const [loading, setLoading] = useState(false);
+	const [email, setEmail] = useState('');
+	const [password, setPassword] = useState('');
+
+	async function signInWithEmail(e: any) {
```
Comment (Thenlie) [inline src/screens/LoginScreen.tsx:9]: `e: any` -> `e: React.SyntheticEvent`

File: src/screens/SignupScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { useState } from 'react';
+import { supabase } from '../helpers/supabaseClient';
+
+export default function SignupScreen() {
+	const [loading, setLoading] = useState(false);
+	const [email, setEmail] = useState('');
+	const [password, setPassword] = useState('');
+
+	const signUp = async (e: any) => {
```
Comment (Thenlie) [inline src/screens/SignupScreen.tsx:9]: `e: any` -> `e: React.SyntheticEvent`

File: src/screens/LoginScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+		e.preventDefault();
+		const { data, error } = await supabase.auth.signInWithPassword({
```
Comment (Thenlie) [inline src/screens/LoginScreen.tsx:11]: Add error handling or `TODO` comment

File: src/screens/SignupScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+		e.preventDefault();
+		const { data, error } = await supabase.auth.signUp({
```
Comment (Thenlie) [inline src/screens/SignupScreen.tsx:11]: Add error handling or `TODO` comment

File: src/screens/SignupScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+							type="email"
+							placeholder="Your email"
+							value={email}
+							onChange={(e) => setEmail(e.target.value)}
+						/>
+						<input
+							id=""
+							className="inputField"
+							type="password"
+							placeholder="Your pasword"
```
Comment (Thenlie) [inline src/screens/SignupScreen.tsx:40]: pasword -> password

File: src/screens/LoginScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+						/>
+						<input
+							id=""
+							className="inputField"
+							type="password"
+							placeholder="Your pasword"
+							value={password}
+							onChange={(e) => setPassword(e.target.value)}
+						/>
+						<button className="button block" aria-live="polite">
```
Comment (Thenlie) [inline src/screens/LoginScreen.tsx:43]: This works, but why does it not have to be `type="submit"` ?

File: src/screens/LoginScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	async function signInWithEmail(e: any) {
+		e.preventDefault();
+		const { data, error } = await supabase.auth.signInWithPassword({
+			email: email,
+			password: password,
+		});
+	}
+
+	return (
+		<div className="row flex-center flex">
```
Comment (Thenlie) [inline src/screens/LoginScreen.tsx:18]: Where are these class names coming from?

File: src/screens/LoginScreen.tsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react';
+import { supabase } from '../helpers/supabaseClient';
+
+export default function LoginScreen() {
+	const [loading, setLoading] = useState(false);
```
Comment (Thenlie) [inline src/screens/LoginScreen.tsx:5]: What is the purpose of this state?

File: src/screens/SignupScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,52 @@
+import { useState } from 'react';
+import { supabase } from '../helpers/supabaseClient';
+
+export default function SignupScreen() {
+	const [loading, setLoading] = useState(false);
```
Comment (Thenlie) [inline src/screens/SignupScreen.tsx:5]: What is the purpose of this state?

**Review by Thenlie — CHANGES_REQUESTED** [review]
Looking good! All seems to work on my end, mostly some syntactical changes and nitpicks, a few questions as well.

File: src/screens/LoginScreen.tsx (hunk 10 above)
Comment (maxh1231) [reply src/screens/LoginScreen.tsx:43]: @Thenlie Default button type is "submit".

submit: The button submits the form data to the server. This is the default if the attribute is not specified for buttons associated with a <form>, or if the attribute is an empty or invalid value.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

**Review by Thenlie — APPROVED** [review]
I left two comments, one is a suggestion for future development, one is an actual change request. I'm approving and fine to change it later, but if you want to fix it, feel free. It could also be a good excuse to practice `git rebase -i` :P

Everything looks awesome though! Nice work 🚀

**Comment by Thenlie** [comment]
The following (acceptable) warnings are generated on the ES Lint build. These will be resolved when error handling is added
```
/home/runner/work/Streamability/Streamability/src/screens/LoginScreen.tsx
  10:11  warning  'data' is assigned a value but never used   @typescript-eslint/no-unused-vars
  10:1[7](https://github.com/Thenlie/Streamability/actions/runs/3414205191/jobs/5681850903#step:4:8)  warning  'error' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/runner/work/Streamability/Streamability/src/screens/SignupScreen.tsx
  [10](https://github.com/Thenlie/Streamability/actions/runs/3414205191/jobs/5681850903#step:4:11):[11](https://github.com/Thenlie/Streamability/actions/runs/3414205191/jobs/5681850903#step:4:12)  warning  'data' is assigned a value but never used   @typescript-eslint/no-unused-vars
  10:17  warning  'error' is assigned a value but never used  @typescript-eslint/no-unused-vars
  ```

**Review by Thenlie — APPROVED** [review]
✅

**Pull request merged**